src/js/stores/MesosStateStore.js

```diff
@@ -179,10 +187,21 @@ class MesosStateStore extends GetSetBaseStore {
     let services = this.get('lastMesosState').frameworks || [];
     let memberTasks = {};
 
-    services.forEach(function (service) {
+    const schedulerTaskIds = services.reduce((memo, service) => {
+      const serviceName = service.name.split('.').reverse().join('/');
+      const schedulerTask = this.getSchedulerTaskFromServiceName(serviceName);
+
+      if (schedulerTask) {
+        memo.push(schedulerTask.id);
+      }
+
+      return memo;
+    }, []);
+
+    services.forEach((service) => {
       service.tasks.forEach(function (task) {
         if (task.slave_id === nodeID) {
-          memberTasks[task.id] = task;
+          memberTasks[task.id] = isSchedulerTask(task, schedulerTaskIds);
```

Comment:
Yeah, this store desperately needs a refactoring and I can't wait to 🔥  it once we have GraphQL in place. However, I wasn't referring to the general problems but to the helper and methods you've introduced.
You added functions to first create a list of scheduler task ids by iterating over the complete list of services. Then we iterate over the same list of service where you add the call to flag tasks as scheduler tasks. By using an array instead of a Map to store the the scheduler ids add even more unnecessary calls.
Please combine the `reduce` and `forech` and change the data structure (do we even need to store the ids?), to improve the performance.